sync.go

```diff
@@ -42,6 +42,10 @@ var (
 	disabled = func(args ...interface{}) string { return fmt.Sprint(args...) }
 )
 
+// Running too many syncs at once can exhaust file descriptor limits.
+// Empirically, ~90 is enough to hit the macOS default limit of 256.
+const maxConcurrentSyncs = 10
```

Comment:
That would be fine as far as the file descriptor limit is concerned, but it seems unlikely to help - as long as you're keeping all the cores busy more processes will just slow things down. Maybe this should be a multiple of runtime.NumCPUs.